client/js/views/visualize.react.jsx

```diff
@@ -91,7 +87,7 @@ var Visualize = React.createClass({
   },
 
   render: function () {
-    if (this.state.commits.length > 0) {
+    if (Object.keys(this.state.fileTree).length > 0) { //commits loaded--this is bad, every commitIndex change this whole thing loads again inc file view
```

Comment:
render blocker this way, or else this tries to render right after the commits are loaded (but the trees and paths are not loaded yet)